submission/distributor.go

```diff
@@ -179,7 +181,22 @@ func incRspsCounter(logURL string, endpoint string, rspErr error) {
 			status = err.StatusCode
 		}
 	}
-	rspsCounter.Inc(logURL, endpoint, strconv.Itoa(status))
+	errCounter.Inc(logURL, endpoint, strconv.Itoa(status))
+}
+
+// incErrCounter increments corresponding errCounter if any error occurred during
+// submission to a Log.
+func incErrCounter(logURL string, endpoint string, rspErr error) {
+	if rspErr == nil {
+		return
+	}
+	if err, ok := rspErr.(client.RspError); ok {
+		if err.Err != nil && err.StatusCode == http.StatusOK {
+			errCounter.Inc(logURL, endpoint, "invalid_sct")
+			return
+		}
+	}
+	errCounter.Inc(logURL, endpoint, "connection_failed")
```

Comment:
The condition under which we get here is: (`rspErr` is not a `client.RspError`) **OR** (it is, but `Err == nil || StatusCode =! StatusOK`). There is this `Err == nil` loophole which doesn't look like "connection_failed". I'd call it "unknown_error" or so.

Also consider:
- distinguishing "unknown_error" from an unknown `RspError`, e.g. increase the counter for "rsp_error" 2 lines above
- not reporting the `Err == nil` and `StatusCode == OK`

Summing this all up, here is my suggestion:

```golang
	switch err, ok := rspErr.(client.RspError) {
	case !ok:
		errCounter.Inc(logURL, endpoint, "unknown_error")
	case err.Err != nil && err.StatusCode == http.StatusOK:
		errCounter.Inc(logURL, endpoint, "invalid_sct")
	case err.Err != nil: // err.StatusCode != http.StatusOK.
		errCounter.Inc(logURL, endpoint, "rsp_error")	
	}
```

Reply:
Done